src/app/modules/main/wallet_section/new_tokens/io_interface.nim

```diff
@@ -0,0 +1,45 @@
+import app_service/service/new_token/service_items
+
+const SIGNAL_MODULE_TOKEN_LIST_UPDATED* = "signalModuleTokenListUpdated"
+const SIGNAL_MODULE_TOKEN_LIST_ABOUT_TO_BE_UPDATED* = "signalModuleTokenListAboutToBeUpdated"
+
+type
+  SourcesOfTokensModelDataSource* = tuple[
```

Comment:
There are several reasons why we proposed this approach.
1. First, we want model to be, in most cases, a lightweight proxy over ground truth - `Service`. The responsibility of models is to expose data in a way which is understandable to Qml world. For that reason we don't want to pass any real data to the model, like sequence in this case. Now our pseudo-interface contains `getSourcesOfTokensList` but even better it could be `getSource(idx: int)` to avoid specifying any concrete type of container.  Then it cannot be refactored to pass source via the ctor directly as in case of sequence if items. Model may want to ask for another information from the service, it can be a single value returned from some service's method. It cannot be passed via ctor up-front.
2. Our pseudo-interface precisely specifies what given model needs to ask about to serve data to UI layer. That interface is dedicated to that model and because of that can be narrow, fulfilling SOLID's Interface segregation principle.
3. Model's taking interface-like object is testable. Not very useful right now but good from architecture point in general. We could easily create mocks to test model in separation, observing if model calls our interface in expected way and forwards data from interface to next layer as intended.

The approach is based on this article: https://www.openmymind.net/Interfaces-In-Nim/. I'm not `nim` expert and I didn't find anything better. If there is better alternative to achieve the same behavior, I'm open to suggestions!

Please consider my answers and please let me know if it makes sense for you :) Thank for your input!

Reply:
I agree and know about all those reasons, but in this particular case, since there is classical object-oriented inheritance in Nim like we have in c++, we won't get too much with this level of interaction. Could we try Nim's concepts instead of this https://www.openmymind.net/Interfaces-In-Nim ?

Reply:
Or anyhow, it will nail down to more or less the same.

Reply:
@saledjenic so would you like to propose some replacement for https://www.openmymind.net/Interfaces-In-Nim based on nim'c concepts or you are ok to continue with the current approach as it's doing basically the same job in the end?

Reply:
@micieslak yes, it's more or less the same, but concepts are Nim's built-in feature and I am more for that, than for this mimic of interfaces.